### DISCUSSION TITLE
Dark mode for PDFs

### DISCUSSION THREAD

**Discussion by NathanZlion** [zen-browser/desktop#3982] [Ideas]
I don't know how many of you read pdfs in zen but I do a lot. I apply night theme and for websites that still appear white I use and extension. So my eyes are safe. Well until I open a pdf. The moment I open a pdf the bright light hits me. It would be great to have a dark mode for for pdf reading. Just like the Document Viewer on my Ubuntu machine has. Here's a photo of the zen document viewer and the same pdf opened in night mode on my device.

![image](https://github.com/user-attachments/assets/4928f198-af77-4bbc-bf7f-cc620d1bdf24)

![image](https://github.com/user-attachments/assets/661b71cf-9bea-4afb-818d-31a337291b12)

As you can see the latter one is much more peaceful.

**Comment by martian0x80** [comment]
I really need this feature, since local files (`file://`) are beyond the reach of browser extensions. I am willing to work on this, if this is accepted. Although, I have no idea where to begin, I'll do some research on where to start.

**Comment by N1N74** [comment]
I hope we're able run custom js on pdf.js level so it's more flexible. This will enable crazy stuff like this:

![](https://raw.githubusercontent.com/belinghy/PDFRefPreview/assets/assets/demo.gif)

from: https://github.com/belinghy/PDFRefPreview 
These projects only works as a bookmarklet and won't work on iframes so supporting custom js with pdf.js at browser level would be great.

---
PS here's a bookmarklet that switches dark mode on pdf:
```
javascript:(function(){document.querySelectorAll('.page,#thumbnailsView').forEach(el=>{if(el.style.filter===''){el.style.filter='grayscale(1)invert(1)contrast(90%)';}else{el.style.filter='';}})})();
```
If you want this permanently, you can modify `userchrome.css` to do that.

**Comment by zhangdajian-sdu** [comment]
I would also appreciate native dark mode for PDF pages in Zen. Dark Reader works for ordinary web pages, but cannot modify the protected built-in PDF viewer.

My main use case is reading research papers with equations, plots, and figures. I would prefer a consistent dark appearance across the entire page, including figures, rather than bright figure regions interrupting a dark page. Inverting or recoloring figures along with the rest of the page is acceptable; preserving their original colors in dark mode is not a requirement.

A useful first version would provide:

- A toolbar toggle between original colors and dark PDF page rendering.
- Consistent treatment of the page background, text, equations, and figures, with readable contrast. A whole-page inversion approach would be acceptable as a starting point.
- Support for both online and local PDFs without custom CSS or an additional viewer extension.
- Display-only changes, leaving the original PDF and its printed/exported colors unchanged.

A quick return to original colors would still be useful when checking a color-sensitive plot, but normal reading should have a unified dark appearance without switching modes for every figure.

[doqment](https://github.com/shivaprsd/doqment) is a PDF.js-based reference implementation with configurable PDF reading colors that may be useful to evaluate.
